File: boilerplate-fullstack-main/boilerplate-fullstack-main/WebApp/src/interfaces/productivity/Activity.ts
export interface ActivityCreatePayload {
  companyId: number;
  activityTypeId: number;
  name: string;
  pointsBase: number;
}

export interface ActivityRead {
  id: number;
  companyId: number;
  activityTypeId: number;
  name: string;
  pointsBase: number;
  active: boolean;
  deleted: boolean;
}

File: boilerplate-fullstack-main/boilerplate-fullstack-main/Api/Models/Activity.cs
namespace Api.Models;

public class Activity
{
    public int Id { get; set; }
    public int CompanyId { get; set; }
    public int ActivityTypeId { get; set; }
    public string Name { get; set; } = string.Empty;
    public int PointsBase { get; set; }
    public bool Active { get; set; } = true;
    public bool Deleted { get; set; } = false;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Company? Company { get; set; }
    public ActivityType? ActivityType { get; set; }
    public ICollection<FiscalActivity> FiscalActivities { get; set; } = new List<FiscalActivity>();
    public ICollection<ServiceOrder> ServiceOrders { get; set; } = new List<ServiceOrder>();
}
